### PR TITLE
Replace string with InputTag in IntTestAnalyzer

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

Also make the module `edm::global`. The change is motivated by me wanting to use product instance names for input products.

#### PR validation:

Framework and `HeterogeneousCore/SonicCore` unit tests pass.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
@wddgit @Dr15Jones any objections?

**Comment by Dr15Jones** [comment]
Seems good to me. The only "wish list" item would be a `fillDescription`

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by wddgit** [comment]
Looks good to me. +1

File: FWCore/Framework/test/stubs/ToyAnalyzers.cc [hunk 1]
```diff
@@ -46,26 +46,33 @@ namespace edmtest {
...
-      iEvent.getByLabel(moduleLabel_, handle);
-      if (handle->value != value_) {
-        throw cms::Exception("ValueMissMatch") << "The value for \"" << moduleLabel_ << "\" is " << handle->value
-                                               << " but it was supposed to be " << value_;
+    void analyze(edm::StreamID, edm::Event const& iEvent, edm::EventSetup const&) const {
+      auto const& prod = iEvent.get(token_);
+      if (prod.value != value_) {
+        edm::ProductLabels labels;
+        labelsForToken(token_, labels);
+        throw cms::Exception("ValueMissMatch")
```
Comment (kpedro88) [inline FWCore/Framework/test/stubs/ToyAnalyzers.cc:67]: while we're here... MissMatch -> Mismatch

**Comment by makortel** [comment]
@cmsbuild, please abort

File: FWCore/Framework/test/stubs/ToyAnalyzers.cc (hunk 1 above)
Comment (makortel) [reply FWCore/Framework/test/stubs/ToyAnalyzers.cc]: Thanks

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by makortel** [comment]
+1

**Comment by qliphy** [comment]
+1

**Pull request merged**